src/setup-julia.ts

```diff
@@ -69,14 +69,21 @@ async function run() {
 
         if (!juliaPath) {
             core.debug(`could not find Julia ${arch}/${version} in cache`)
-            const juliaInstallationPath = await installer.installJulia(versionInfo, version, arch)
 
-            // Add it to cache
-            juliaPath = await tc.cacheDir(juliaInstallationPath, 'julia', version, arch)
+            // https://github.com/julia-actions/setup-julia/pull/196
+            // we want julia to be installed with unmodified file mtimes
+            // but tc.cacheDir uses `cp` which destroys mtime
+            // and `tc` provides no API to get the tool directory alone
+            // so hack it by installing a dummy julia file then use the path it returns
+            // and extract the archives directly to that location
+            const tempDummyDir = fs.mkdtempSync('julia-dummy-')
+            juliaPath = await tc.cacheDir(tempDummyDir, 'julia', version, arch)
+            await installer.installJulia(juliaPath, versionInfo, version, arch)
```

Comment:
Should be equivalent but uses the more descriptive temporary directory name:
```suggestion
            const tempDir = fs.mkdtempSync(`julia-${arch}-${version}-`)
            juliaInstallDir = await tc.cacheDir(tempDir, 'julia', version, arch)
            await installer.installJulia(juliaInstallDir, versionInfo, version, arch)
```

Reply:
Naming it like that makes it seem like it has to be a special name. It doesn't, just needs to be a temp dir.

As for renaming `juliaPath` that won't work as it's a variable defined and used outside of this scope

Reply:
Upon further reflection I agree with your points. I do think we should just name this `emptyDir` to make it clearer that this directory is empty and will always be empty:
```suggestion
            const emptyDir = fs.mkdtempSync('empty')
            juliaPath = await tc.cacheDir(emptyDir, 'julia', version, arch)
            await installer.installJulia(juliaPath, versionInfo, version, arch)
```